knative/deployer.go

```diff
@@ -219,6 +229,19 @@ func updateService(f fn.Function, newEnv []corev1.EnvVar, newEnvFrom []corev1.En
 		for k, v := range f.Annotations {
 			service.ObjectMeta.Annotations[k] = v
 		}
+		// I hate that we have to do this. Users should not see these values.
+		// It is an implementation detail. These health endpoints should not be
+		// a part of func.yaml since the user can only mess things up by changing
+		// them. Ultimately, this information is determined by the language pack.
+		// Which is another reason to consider having a global config to store
+		// some metadata which is fairly static. For example, a .config/func/global.yaml
```

Comment:
Per my earlier comment, health endpoints may be better solved by providing sane defaults in `deployer.go` directly.  However, this is a good time to drop 2c on the global config.  In general I think this is a very good idea and a good way forward.

We can take the `git` configuration system as a guide here I think:

There are many settings which we default for the Function developer.   Making guesses on their behalf being preferred over prompting for everything.

There exist certain settings where the default chosen presents a nontrivial chance of the user desiring to modify that value.  This can be accomplished by the user modifying their local Function's `.func.yaml` either manually, via the CLI's `config` subcommand, or via the client library if this is an integration situation.

In the event this change to the default value is something the user would like to happen by default on all Functions created by the current user, they can make the change in their global `$XDG_CONFIG_HOME/func/config.yaml` (or via the CLI by providing the `--global` flag on config changes).

This seems like a sane approach; has worked well for git, and since we expect most users to be familiar with git already, this reuses their knowledge; requiring little cognitive load to assimilate.